src/os/bluestore/BlueStore.cc

```
@@ -19072,67 +19072,70 @@ int BlueStore::read_allocation_from_drive_for_bluestore_tool(bool test_store_and
+    stats.insert_count = 0;
+    auto count_entries = [&](uint64_t extent_offset, uint64_t extent_length) {
+      stats.insert_count++;
+    };
+    allocator->dump(count_entries);
+    dout(1) << "\n" << " <<<FINISH>>> in " << duration << " seconds; insert_count=" << stats.insert_count << dendl;
+    dout(1) << "\n" << " <<<FINISH>>> in " << duration << " seconds; extent_count=" << stats.extent_count << dendl;
 
-  dout(5) << "calling compare_allocator(alloc) insert_count=" << stats.insert_count << dendl;
-  ret = compare_allocators(allocator, alloc, stats.insert_count, memory_target);
-  if (ret == 0) {
-    dout(5) << "SUCCESS!!! compare(allocator, alloc)" << dendl;
-  } else {
-    derr << "**** FAILURE compare(allocator, alloc)::ret=" << ret << dendl;
+    ret = compare_allocators(allocator.get(), alloc, stats.insert_count, memory_target);
+    if (ret == 0) {
+      dout(5) << "SUCCESS!!! compare(allocator, alloc)" << dendl;
+    } else {
+      derr << "**** FAILURE compare(allocator, alloc)::ret=" << ret << dendl;
+    }
   }
 
   if (test_store_and_restore) {
     _close_db_leave_bluefs();
     dout(5) << "calling store_allocator(alloc)" << dendl;
     store_allocator(alloc);
-    Allocator* alloc2 = create_bitmap_allocator(bdev_size);
+    auto alloc2 = unique_ptr<Allocator>(create_bitmap_allocator(bdev->get_size()));
     if (alloc2) {
-      dout(5) << "bitmap-allocator=" << alloc2 << dendl;
+      dout(5) << "bitmap-allocator=" << alloc2.get() << dendl;
       dout(5) << "calling restore_allocator()" << dendl;
       uint64_t num, bytes;
-      int ret = restore_allocator(alloc2, &num, &bytes);
+      int ret = restore_allocator(alloc2.get(), &num, &bytes);
       if (ret == 0) {
 	// add allocation space used by the bluefs itself
-	ret = add_existing_bluefs_allocation(alloc2, stats);
+	ret = add_existing_bluefs_allocation(alloc2.get(), stats);
 	if (ret < 0) {
-	  _close_db_and_around();
+	  //_close_db_and_around();
 	  return ret;
```

And this is out-of-pattern.
Why in this case we silently return?